pkg/controller/scheduler.go

```diff
@@ -758,6 +758,14 @@ func (c *Controller) shouldAdvance(canary *flaggerv1.Canary, canaryController ca
 		return true, nil
 	}
 
+	// Make sure to sync lastAppliedSpec even if the canary is in a failed state.
+	if canary.Status.Phase == flaggerv1.CanaryPhaseFailed {
```

Comment:
Then `LastAppliedSpec` refers to the primary, If the canary fails then why are we updating the hash, wouldn't this break the rollback no-op after the initialised phase?

Reply:
If the canary is in a failed state, with the primary deploy at `6.0` and canary deploy at `6.1`, and then we update the canary deploy to be at `6.0`, the `lastAppliedSpec` should reflect that. If it doesn't reflect that, `lastAppliedSpec` would still be the hash calculated from the canary deploy spec with image version `6.1`, which will prevent the canary from getting triggered after we update the canary deploy from `6.0` to `6.1`.

Reply:
Got it! thanks